File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class TestClass
{
    public void TestMethod()
    {
        var result = new List<string> { "y" };
        var map = new Dictionary<string, int>
        {
            { "x", 5 }
        };
        var keys = map.Keys.ToArray();
        Console.WriteLine(result[0]);
        Console.WriteLine(keys[0]);
    }
}

public class Program
{
    static public void Main(string[] args)
    {
        try 
        {
            new TestClass().TestMethod();
        }
        catch (System.Exception e)
        {
            System.Console.WriteLine($"Exception: {e.Message}");
        }
    }
}

File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.go
package main

import "fmt"

type TestClass struct {
}

func NewTestClass() *TestClass {
    this := new(TestClass)
    return this
}

func (this *TestClass) TestMethod() {
    result := []string{"y"}
    map := map[string]int{
      "x": 5,
    }
    keys := make([]string, 0, len(map))
    for  key, _ := range map {
      keys = append(keys, key)
    }
    fmt.Println(result[0])
    fmt.Println(keys[0])
}

func main() {
    defer func() {
      if r := recover(); r != nil {
          fmt.Print("Exception: ", r)
      }
    }()

    c := NewTestClass()
    c.TestMethod();
}